podcasts/Constants.swift

```diff
@@ -298,6 +310,56 @@ struct Constants {
     }
 
     static let defaultDebounceTime: TimeInterval = 0.5
+
+    /// The `SettingValue` provides a way to:
+    ///     - Define a UserDefaults setting
+    ///     - Specify the default value to be used if it's not set
+    ///     - Retrieve and save the value to the user defaults
+    /// Example:
+    /// let helloWorld = SettingValue("hello", defaultValue: "world")
+    /// print(helloWorld.value()) -> "world" // Default value is used
+    /// helloWorld.save("hello world")
+    /// print(helloWorld.value()) -> "hello world" // Custom value is used
+    ///
+    /// This uses Generics and the `defaultValue` to define the value type that should be set and returned.
+    /// If you want to provide a nullable value you can:
+    ///
+    /// Specify an optional and a default value by wrapping the value in an `Optional`
+    ///
+    ///     SettingValue("nullable", defaultValue: Optional("HelloWorld"))
+    ///
+    /// Specify a type and a default value of nil by using `.none`
+    ///
+    ///     SettingValue("nullable", defaultValue: String?.none)
+    ///
+    struct SettingValue<Value> {
+        let key: String
+        let defaultValue: Value
+
+        init(_ key: String, defaultValue: Value) {
+            self.key = key
+            self.defaultValue = defaultValue
+        }
+
+        /// Retrieve the stored value from the UserDefaults or the `defaultValue` if there isn't a stored value
+        func value(in defaults: Foundation.UserDefaults = .standard) -> Value {
```

Comment:
Having the `defaults` as a function argument might open this class to use `save` with one UserDefaults and `value()` with another.

What do you think about having the `UserDefaults` in `init` and changing `value` to be a property?

I think we will eliminate the chance of different defaults happening plus it will be a bit more Swift.

Reply:
Makes sense, done here: 91c8d5e77934db58ccf57c9ab3c8a40eb53196e3